packages/app/src/cli/commands/app/release.ts

```diff
@@ -54,6 +60,11 @@ export default class Release extends Command {
 
     const specifications = await loadLocalExtensionsSpecifications(this.config)
     const app: AppInterface = await loadApp({specifications, directory: flags.path, configName: flags.config})
+
+    const requiredNonTTYFlags = ['force']
+    if (!apiKey && !app.configuration.client_id) requiredNonTTYFlags.push('client-id')
+    this.failMissingNonTTYFlags(flags, requiredNonTTYFlags)
```

Comment:
Note this might break existing flows... but the error message should be clear enough, and it's the right thing to do long-term. So I think this makes sense.

Also a bit of data wrangling suggests no one is actually using this in CI yet... maybe because of the lack of a `--force` flag!